scipy/optimize/tests/test_bracket.py

```diff
@@ -345,6 +345,35 @@ def f(x):
                                 xmin=1)
         assert not res.success
 
+        # 5. Edge case. Multiple searches in same direction terminate
```

Comment:
I wrote these tests before I understood what was going on. I guess these aren't even edge cases, just straight up bugs we missed. I'll separate these out into a different test since they're not special cases.

Reply:
Actually, nevermind. I guess these new tests are fine here. I'm not sure where to put them or how to name them otherwise.

Reply:
Could create a new bug fix test. That would be fine with me.